server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1160,62 +1160,259 @@ public void apply(T value, Settings current, Settings previous) {
         return new Setting<>(key, s -> Integer.toString(defaultValue.id), s -> Version.fromId(Integer.parseInt(s)), properties);
     }
 
+    private static boolean isFiltered(Property[] properties) {
+        return properties != null && Arrays.asList(properties).contains(Property.Filtered);
+    }
+
+    // Float
+
+    private static float parseFloat(String s, float minValue, float maxValue, String key, boolean isFiltered) {
+        float value = Float.parseFloat(s);
+        if (value < minValue) {
+            String err = "Failed to parse value" + (isFiltered ? "" : " [" + s + "]") + " for setting [" + key + "] must be >= " + minValue;
+            throw new IllegalArgumentException(err);
+        }
+        if (value > maxValue) {
+            String err = "Failed to parse value" + (isFiltered ? "" : " [" + s + "]") + " for setting [" + key + "] must be <= " + maxValue;
+            throw new IllegalArgumentException(err);
+        }
```

Comment:
I think it would be better to phrase the error message to include both the min and max:

`Failed to parse value: <key> must in in the range [<min>, <max>]`

It's a better user experience when all the constraints are spelled out. Also, less duplication in this code.

Reply:
The problem is that often max is MAX_INT or MAX_LONG, so in those cases you don't want this.